### PR TITLE
Feat/improve samm mapping

### PR DESCRIPTION
**Pull request by eschrewe**
## Description
* ProductStockSammDto now supports MeasurementUnit enum instead of String representation
* Mapping Annotation added to MeasurementUnit
* Stock and its sub classes now support Sites and Addresses
* Incoming data from ProductStockSammDto's will now be merged with other PartnerProductStocks if locationId, locationIdType and MeasurementUnit are matching

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java [hunk 1]
```diff
@@ -107,69 +100,46 @@ public ProductStockSammDto toSamm(Stock stock) {
...
-                foundMaterialDto,
-                quantity,
-                measurementUnit,
-                atSiteBpns,
-                supplierPartner,
-                lastUpdatedOnDate
-        );
+    public List<PartnerProductStock> sammToPartnerProductStocks(ProductStockSammDto samm, Partner partner) {
+        ArrayList<PartnerProductStock> output = new ArrayList<>();
+        Material material = materialService.findByOwnMaterialNumber(samm.getMaterialNumberCustomer());
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java:120]: This logic is not appropriate. If Catena-X number is given, then this number must be used. It's like a cascade (use if given):
- catenaX
- supplier
- customer

Customer number is mandatory, rest is optional :)

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java [hunk 2]
```diff
@@ -107,69 +100,46 @@ public ProductStockSammDto toSamm(Stock stock) {
...
 
-
     /**
-     * Utility method to deserialize a ProductStockSammDto on the side of the customer
+     * Utility method to deserialize a ProductStockSammDto on the side of the customer.
+     * Since each ProductStockSammDto may contain multiple Positions, this method will
+     * return a list of PartnerProductStocks, where each Samm-Position is mapped to an
+     * instance of PartnerProductStock.
+     *
+     * The caller of this method has to decide how whether he wants to aggregate some of
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java:109]: Please add a note, that orderPositionReferences are currently ignored.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Great effort. Please incorporate reviews + merge old branch + rerun dependencies check for both, frontend and backend.

**Comment by tom-rm-meyer-ISST** [comment]
@eschrewe plese merge the conflicts for the backend dependencies with the main branch

**Comment by eschrewe** [comment]
Done

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java [hunk 3]
```diff
@@ -107,69 +100,75 @@ public ProductStockSammDto toSamm(Stock stock) {
...
-                        .stream().findFirst().get().getSupplierStockLocationId().getLocationId();
-
-        PartnerDto supplierPartner = modelMapper.map(partner, PartnerDto.class);
+    public List<PartnerProductStock> sammToPartnerProductStocks(ProductStockSammDto samm, Partner partner) {
+        ArrayList<PartnerProductStock> output = new ArrayList<>();
+        Material material = null;
+        if(samm.getMaterialNumberCatenaX().isPresent()) {
+            material = materialService.findByMaterialNumberCx(samm.getMaterialNumberCatenaX().get());
+        }
+        if(material == null && samm.getMaterialNumberCustomer() != null) {
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java:127]: The cascade of which material number must be used must be as follows:
- catena-x
- supplier
- customer

Please switch

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Please switch order of customer material number and supplier material number during search.

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java (hunk 3 above)
Comment (eschrewe) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java]: Done!

**Review by tom-rm-meyer-ISST — APPROVED** [review]
Great work! LGTM, Quality check is related to helm chart structure

**Pull request merged**